pkg/controller/infrastructure/configvalidator.go

```diff
@@ -70,6 +74,18 @@ func (c *configValidator) Validate(ctx context.Context, infra *extensionsv1alpha
 	// Validate infrastructure config
 	logger.Info("Validating infrastructure configuration")
 	allErrs = append(allErrs, c.validateFloatingPoolName(ctx, networkingClient, config.FloatingPoolName, field.NewPath("floatingPoolName"))...)
+	if config.Networks.ID != nil {
+		allErrs = append(allErrs, c.validateNetwork(ctx, networkingClient, *config.Networks.ID, field.NewPath("networks.id"))...)
+	}
+	if config.Networks.SubnetID != nil {
+		allErrs = append(allErrs, c.validateSubnet(ctx, networkingClient, *config.Networks.SubnetID, *config.Networks.ID, field.NewPath("networks.subnetId"))...)
+	}
+	if config.Networks.Router != nil && config.Networks.Router.ID != "" {
```

Comment:
Imo it's unnecessary to check for `config.Networks.Router.ID != ""` because why should you be able to set the router config without an ID in the first place.

Reply:
This is a bit contradicting with the comment above. The controller that validates that you have a subnet.id iff a network.id exists is the same that performs the validation about the empty router.id.

Reply:
fair point, I'm fine with leaving it as is.